src/context/mod.rs

```diff
@@ -55,6 +56,15 @@ impl CallbackInfo<'_> {
         }
     }
 
+    #[cfg(feature = "napi-runtime")]
+    fn kind<'b, C: Context<'b>>(&self, cx: &C) -> CallKind {
```

Comment:
We should plan some time to discuss when references to `Context` should be mutable and when they shouldn't. I.e., are there places we are relying on the exclusive reference to ensure correctness?

Some n-api, but not all, will mutate the `Env` (e.g. changing state, throwing), but maybe those should be considered interior mutability and we should still consider that `&Env`?

This code relies on it being a shared reference because `info` is borrowing `self` while also passing it to `kind`.

This comment is kind of long and only partially relevant. I'll put more detail notes in an issue.